internal/resolution/variablesources/installed_package.go

```diff
@@ -76,6 +84,18 @@ func MakeInstalledPackageVariables(
 // Must not return installed bundle as a successor
 type successorsFunc func(allBundles []*catalogmetadata.Bundle, installedBundle *catalogmetadata.Bundle) ([]*catalogmetadata.Bundle, error)
 
+func successorsFuncForOperator(operator operatorsv1alpha1.Operator) successorsFunc {
+	if operator.Spec.UpgradeConstraintPolicy == operatorsv1alpha1.UpgradeConstraintPolicyIgnore {
+		return ignoreConstraints
```

Comment:
Instead of building a new upgrade edge constraint that says "you can upgrade from this bundle to anything", can we just _not_ add the constraint for the this operator in the first place?

Essentially, the behavior in the `Ignore` case should be identical to the "not yet installed" case, which you handle in line 36-40. I'd propose modifying that if statement to also check for `operator.Spec.UpgradeConstraintPolicy == operatorsv1alpha1.UpgradeConstraintPolicyIgnore`

Reply:
🤦‍♂️ Totally forgot that we were going to go this way. Thanks for the reminder. I updated the PR (still in progress) with this appraoch.